### PR TITLE
Implement `CODEOWNERS` corrections

### PR DESCRIPTION
**Pull request by Repiteo**
- Followup to #97866

`CODEOWNERS` has been a bit too overzealous in catching review groups, so this PR mitigates that with two changes:
- Revert adding `_editor` reviewers to `editor` folders in modules. Might revisit in the future, but it's more granular than expected.
- Add underscore before wildcard for `servers`, preventing false positives.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (AThousandShips) [inline .github/CODEOWNERS:236]: I think we need specifics for just the base level cases here, this change will make `/servers/rendering` no longer be owned, nor will `xr`, `audio`, or `text`

So we might want to restore this to the state before #97866 to start with and then redo this in a future PR, so:
```suggestion
/servers/audio*                               @godotengine/audio
/servers/camera*                              @godotengine/xr
/servers/display_server.*                     @godotengine/_platforms
/servers/navigation_server*.*                 @godotengine/navigation
/servers/physics*                             @godotengine/physics
/servers/rendering*                           @godotengine/rendering
/servers/text_server.*                        @godotengine/gui-nodes
/servers/xr*                                  @godotengine/xr
```

**Review by AThousandShips — COMMENTED** [review]
For the meta-teams like `@godotengine/_editor` we might want to look at diversifying the teams in that, and we might want to remove some of the other meta-teams if they aren't working well

But looks good for that part, holding off approval until the servers part has been resolved

CC @bruvzg since you had some thoughts on the wildcard cases in the previous PR

File: .github/CODEOWNERS
Comment (bruvzg) [reply .github/CODEOWNERS:236]: `/servers/xxx*` will not include subdirectories, so we might want two lines for each:

https://git-scm.com/docs/gitignore#_pattern_format

> An asterisk "*" matches anything except a slash.
> A trailing "/**" matches everything inside.

```
/servers/text_*                             @godotengine/gui-nodes
/servers/text/**                            @godotengine/gui-nodes
```

**Comment by dalexeev** [comment]
By the way:

1\. Add missing `.` in line 3:

https://github.com/godotengine/godot/blob/b3bcb2dc14691f7729984128dca26a844f662fa1/.github/CODEOWNERS#L1-L3

2\. Add missing `/` in line 156:

https://github.com/godotengine/godot/blob/b3bcb2dc14691f7729984128dca26a844f662fa1/.github/CODEOWNERS#L155-L157

File: .github/CODEOWNERS
Comment (AThousandShips) [reply .github/CODEOWNERS:236]: Agreed, was mainly wanting to fix this directly and then discuss more in detail the specifics in a follow-up, but if this works I'd say to go with that suggestion

Comment (Repiteo) [reply .github/CODEOWNERS:236]: Combining the two should work for now

**Review by AThousandShips — APPROVED** [review]
Should work well now I think

**Pull request merged**
